### ISSUE TITLE
Add "set" prop to useBoolean

### ISSUE THREAD

**Issue by mateja176** [alibaba/hooks#620]
```ts
const [value, { setTrue, setFalse, toggle }] = useBoolean();
// toggle has a signature of (value?: boolean | undefined) => void
```

The issue is that if you want to pass toggle as an event handler prop i.e. `<button onClick={toggle}>Toggle</button>`, you can't because this results in a type error, or worse a button which can set the value to true only.

It would be perfectly in line with `setTrue` and `setFalse` to also have a `set` prop which would also alleviate responsibility from the `toggle` prop.

Moreover, the philosophy `ahooks` is improve DX, so instead of having to do:

```tsx
const [value, { setTrue, setFalse, toggle }] = useBoolean();
const toggleValue = React.useCallback(() => {
  toggle();
}, [toggle])

<button onClick={toggle}>Toggle</button>
```

With the advent of `set`, you'd be able to write:

```tsx
const [value, { setTrue, setFalse, set, toggle }] = useBoolean();

<button onClick={toggle}>Toggle</button>
```

**Comment by brickspert** [comment]
I very much agree with you.
but we can not change toggle behavior , because it's a break change.

i'm very sad.

**Comment by mateja176** [comment]
Thanks mate. That's understandable, however the change may be part of the version 3 release.

**Comment by brickspert** [comment]
Hello：

这个问题我们已经在 v3.0 版本彻底修复，目前 v3.0 版本已经发布 alpha 版本。

欢迎试用。

文档：https://ahooks-next.surge.sh/
源码：https://github.com/alibaba/hooks/tree/next

安装最新分支
```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```

We have fixed this problem in the v3.0 version, and the alpha has been released. 

Welcome to try.

Doc：https://ahooks-next.surge.sh/
Source Code：https://github.com/alibaba/hooks/tree/next

Install v3.0

```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```
